### ISSUE TITLE
Is `xcode-select --install` step still required?

### ISSUE THREAD

**Issue by oxyc** [geerlingguy/mac-dev-playbook#50]
### Issue type

- Support request

### Summary

Is it still a requirement to manually install Apple's command line tools as the README suggests or was this fixed in https://github.com/geerlingguy/ansible-role-homebrew/pull/53? I'm not entirely clear if it's the same command line tools as the tasks performed are very different...

**Comment by geerlingguy** [comment]
Yeah... I'm not sure, as I don't remember having done that on my most recent fresh install (on my late 2016 MacBook Pro). If someone can confirm it's not required, we can yank it.

**Comment by oxyc** [comment]
ping @elliotweiser

**Comment by elliotweiser** [comment]
That command is no longer required/used upstream. The latest version of the [osx-command-line-tools role](elliotweiser/ansible-osx-command-line-tools) checks that the OS X command line tools are (properly) installed by performing a `pkgutil` and `g++` commands instead.

**Comment by elliotweiser** [comment]
My understanding is that the OSX command line tools (but not necessarily Xcode) are a hard dependency of Homebrew. Is that no longer the case?

**Comment by elliotweiser** [comment]
(Now that I've had more time to process your question, I deleted my earlier comments since I think they obfuscate the real answer.)

Is the machine you are provisioning also the Ansible control node? If so, yes. Otherwise, it is automated for you (as of geerlingguy/ansible-role-homebrew#53). 

If you are using a Mac as your Ansible control node, then the OS X Command Line tools _must be installed first_. Ansible has certain dependencies that can only be downloaded/compiled if the command line tools are installed (e.g. `git` and `gcc`). Using Ansible to self-provision a Mac is a chicken-and-egg problem, but is resolved by running `xcode-select --install`, which incidentally is one of many ways to download/install the command line tools.

However, if you are provisioning a Mac that is _not_ the control node, then you can get away with not installing the command line tools manually. The only system requirement for a managed node is Python 2.6+, which comes for free on a Mac. To clarify, geerlingguy/ansible-role-homebrew#53 only applies to this use-case.

TL;DR: if you want to use Ansible on a Mac, you have to install the OS X Command Line tools first.

More information about installation instructions, control nodes, and managed nodes:

http://docs.ansible.com/ansible/intro_installation.html

**Comment by oxyc** [comment]
Thanks a lot for the great explanation! Now I'm up to speed. Then I guess the order of the installation steps actually need to be updated with `xcode-select --install` prior to installing ansible.

**Comment by geerlingguy** [comment]
Fixed via above PR, thanks @elliotweiser for that explanation, my memory is a bit fuzzy on it.
